fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * MetaDataException.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.RecordCoreException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Exception thrown when there is a problem in {@link com.apple.foundationdb.record.provider.foundationdb.OnlineIndexer} during indexing.
+ */
+@SuppressWarnings("serial")
+@API(API.Status.INTERNAL)
+public class OnlineIndexerException extends RecordCoreException {
```

Comment:
Do you mean the backward compatible OnlineIndexer.RecordBuiltRangeException?

Reply:
No, I'm saying that, I think that there should be exceptions that can get thrown by an indexer that are retriable for various definitions of retriable. For example, timing out on something, or hitting a limit, or having too many transaction conflicts. Others are probably not retriable, like Uniqueness violations, or maybe RecordBuiltRangeException

Reply:
I think I can see your point - it would be nice and clean to have two types of indexer exception - retriable and not. But since this PR is already a monster, maybe this change should be left for another PR? The existing logic and exceptions hadn't changed.